tutorials/external-metastore-for-dataproc/index.md

```diff
@@ -0,0 +1,278 @@
+
+In Cloud Shell, set the default Compute Engine zone and region where you are going to create your Dataproc clusters:
+
+        export PROJECT="$(gcloud info --format='value(config.project)')"
+        export REGION="us-central1"
+        export ZONE="us-central1-a"
+        export WAREHOUSE_MULTI_REGION="us"
+        gcloud config set compute/zone "${ZONE}"        
+
+## Creating resources
+
+The tutorial uses following resources:
+
+ * A Cloud Storage bucket as the data warehouse for test data storage
+ * A cloud SQL instance
+
+### Create the warehouse Cloud Storage bucket
+
+Create a Cloud Storage bucket for storing test data, run the following command on Cloud Shell:
+
+        gsutil mb -p ${PROJECT} -l ${WAREHOUSE_MULTI_REGION} "gs://${PROJECT}-warehouse" 
+
+### Create Cloud SQL instance
+
+
+Create a MySQL on Cloud SQL instance to be used later to host the Hive metastore database.
+
+In Cloud Shell, execute the following command to create a new Cloud SQL instance:
+        
+        gcloud sql instances create hive-metastore-db \
+        --database-version="MYSQL_5_7" \
+        --activation-policy=ALWAYS \
+        --zone ${ZONE}
+
+This command might take a few minutes to complete.
+
+### Create Dataproc clusters for Hive metastore
+
+Create the first single-node Dataproc cluster that will be the Hive metastore service provider for worker clusters.
+
+In Cloud Shell, execute the following command:
+
+        gcloud dataproc clusters create hive-metastore1 \
+        --enable-component-gateway \
+        --scopes sql-admin \
+        --region ${REGION} \
+        --zone ${ZONE} \
+        --single-node \
+        --master-machine-type n1-standard-8 \
+        --master-boot-disk-type pd-ssd \
+        --master-boot-disk-size 500 \
+        --image-version 2.0.29-debian10 \
+        --properties hive:hive.metastore.warehouse.dir=gs://${PROJECT}-warehouse/datasets \
+        --initialization-actions gs://goog-dataproc-initialization-actions-${REGION}/cloud-sql-proxy/cloud-sql-proxy.sh \
+        --metadata hive-metastore-instance=${PROJECT}:${REGION}:hive-metastore-db
+
+This command will take a few minutes to create and initialize the cluster.
+
+Create a failover metastore service cluster using the following command in Cloud Shell:
+
+        gcloud dataproc clusters create hive-metastore2 \
+        --async
+        --enable-component-gateway \
+        --scopes sql-admin \
+        --region ${REGION} \
+        --zone ${ZONE} \
+        --single-node \
+        --master-machine-type n1-standard-8 \
+        --master-boot-disk-type pd-ssd \
+        --master-boot-disk-size 500 \
+        --image-version 2.0.29-debian10 \
+        --properties hive:hive.metastore.warehouse.dir=gs://${PROJECT}-warehouse/datasets \
+        --initialization-actions gs://goog-dataproc-initialization-actions-${REGION}/cloud-sql-proxy/cloud-sql-proxy.sh \
+        --metadata hive-metastore-instance=${PROJECT}:${REGION}:hive-metastore-db
+
```

Comment:
This can't be done, because the Hive metastore cluster creation will fail if the CloudSQL instance is not accessible, and similarly, for Worker Cluster, creation will fail if metastore cluster is not accessible.

Reply:
How about providing a terraform script instead?

Reply:
Great point, you can disregard this suggestion.   (I think when I tested it, I had already created the Cloud SQL instance without --async)